src/Microsoft.OData.Core/Json/JsonValueUtilsAsync.cs

```diff
@@ -0,0 +1,491 @@
+                // to use XmlConvert in all other cases, except infinity
+                string valueToWrite = XmlConvert.ToString(value);
+
+                await writer.WriteAsync(valueToWrite).ConfigureAwait(false);
+                if (valueToWrite.IndexOfAny(DoubleIndicatingCharacters) < 0)
+                {
+                    await writer.WriteAsync(".0").ConfigureAwait(false);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Write a Guid value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">Guid value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, Guid value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteQuotedAsync(value.ToString()).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Write a decimal value
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">Decimal value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, decimal value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteAsync(value.ToString(CultureInfo.InvariantCulture)).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Write a DateTimeOffset value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">DateTimeOffset value to be written.</param>
+        /// <param name="dateTimeFormat">The format to write out the DateTime value in.</param>
+        [SuppressMessage("Microsoft.Globalization", "CA1303:Do not pass literals as localized parameters", MessageId = "Microsoft.OData.Json.JsonValueUtils.WriteQuotedAsync(System.IO.TextWriter,System.String)", Justification = "Constant defined by the JSON spec.")]
+        internal static async Task WriteValueAsync(this TextWriter writer, DateTimeOffset value, ODataJsonDateTimeFormat dateTimeFormat)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            Int32 offsetMinutes = (Int32)value.Offset.TotalMinutes;
+
+            switch (dateTimeFormat)
+            {
+                case ODataJsonDateTimeFormat.ISO8601DateTime:
+                    {
+                        // Uses the same format as DateTime but with offset:
+                        // jsonDateTime= quotation-mark
+                        //  YYYY-MM-DDThh:mm:ss.sTZD
+                        //  [("+" / "-") offset]
+                        //  quotation-mark
+                        //
+                        // offset = 4DIGIT
+                        string textValue = XmlConvert.ToString(value);

```

Comment:
Add CultureInfo

Reply:
I can see you you are suppressing this.

Reply:
@KenitoInc `System.Xml.XmlConvert.ToString(DateTimeOffset)` will do the expected thing here for ISO8601 date format

`System.Xml.XmlConvert.ToString(DateTimeOffset.Now)` => `2021-02-25T17:31:58.9858279+03:00`